src/envelope.js

```diff
@@ -48,852 +48,858 @@ import TimelineSignal from 'Tone/signal/TimelineSignal.js';
- *  again will override the initial add() with new values.
- *
- *  @method  add
- *  @for p5.Envelope
- *  @param {Number} number Constant number to add
- *  @return {p5.Envelope} Envelope Returns this envelope
- *                                     with scaled output
- */
-p5.Envelope.prototype.add = function (num) {
-  var add = new Add(num);
-  var thisChain = this.mathOps.length;
-  var nextChain = this.output;
-  return p5.prototype._mathChain(this, add, thisChain, nextChain, Add);
-};
+  // get rid of the oscillator
+  dispose() {
+    // remove reference from soundArray
+    var index = p5sound.soundArray.indexOf(this);
+    p5sound.soundArray.splice(index, 1);
 
-/**
- *  Multiply the p5.Envelope's output amplitude
- *  by a fixed value. Calling this method
- *  again will override the initial mult() with new values.
- *
- *  @method  mult
- *  @for p5.Envelope
- *  @param {Number} number Constant number to multiply
- *  @return {p5.Envelope} Envelope Returns this envelope
- *                                     with scaled output
- */
-p5.Envelope.prototype.mult = function (num) {
-  var mult = new Mult(num);
-  var thisChain = this.mathOps.length;
-  var nextChain = this.output;
-  return p5.prototype._mathChain(this, mult, thisChain, nextChain, Mult);
-};
-
-/**
- *  Scale this envelope's amplitude values to a given
- *  range, and return the envelope. Calling this method
- *  again will override the initial scale() with new values.
- *
- *  @method  scale
- *  @for p5.Envelope
- *  @param  {Number} inMin  input range minumum
- *  @param  {Number} inMax  input range maximum
- *  @param  {Number} outMin input range minumum
- *  @param  {Number} outMax input range maximum
- *  @return {p5.Envelope} Envelope Returns this envelope
- *                                     with scaled output
- */
-p5.Envelope.prototype.scale = function (inMin, inMax, outMin, outMax) {
-  var scale = new Scale(inMin, inMax, outMin, outMax);
-  var thisChain = this.mathOps.length;
-  var nextChain = this.output;
-  return p5.prototype._mathChain(this, scale, thisChain, nextChain, Scale);
-};
-
-// get rid of the oscillator
-p5.Envelope.prototype.dispose = function () {
-  // remove reference from soundArray
-  var index = p5sound.soundArray.indexOf(this);
-  p5sound.soundArray.splice(index, 1);
-
-  this.disconnect();
-  if (this.control) {
-    this.control.dispose();
-    this.control = null;
+    this.disconnect();
+    if (this.control) {
+      this.control.dispose();
+      this.control = null;
+    }
+    for (var i = 1; i < this.mathOps.length; i++) {
+      this.mathOps[i].dispose();
+    }
   }
-  for (var i = 1; i < this.mathOps.length; i++) {
-    this.mathOps[i].dispose();
+}
+export class Env extends Envelope {
+  constructor(t1, l1, t2, l2, t3, l3) {
+    console.warn(
+      'WARNING: p5.Env is now deprecated and may be removed in future versions. ' +
+        'Please use the new p5.Envelope instead.'
+    );
+    super();
```

Comment:
pass the arguements of constructor to super() too.
We have a different aproach to handle the deprecations now.
please put this ENV to src/deprecations .